components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/internal/I18nMgtServiceComponent.java

```diff
@@ -97,9 +97,10 @@ protected void activate(ComponentContext context) {
                 log.error("Error registering Email Template Mgt Service.");
             }
 
+            NotificationTemplateManager notificationManager = new EmailTemplateManagerImpl();
             // Register EmailTemplateManagerImpl.
             ServiceRegistration notificationManagerSR = bundleCtx
-                    .registerService(NotificationTemplateManager.class.getName(), emailTemplateManager, null);
+                    .registerService(NotificationTemplateManager.class.getName(), notificationManager, null);
```

Comment:
EmailTemplateManagerImpl implementes both EmailTemplateManager and NotificationTemplateManager interfaces. Does it makes sense anymore?

Reply:
Yes.. As of now these both needs to be implemented since `NotificationTemplateManager` don't have enough functionalities to deprecate the `EmailTemplateManager` and `EmailTemplateManager` is not general enough to handle different notification channel types.

This is mainly due to NotificationTemplate management is not fully complete. We will have to work on https://github.com/wso2/product-is/issues/9762 and get `NotificationTemplateManager` improved to have all needed functionalities and retire `EmailTemplateManager`..

At that point it should be possible to `EmailTemplateManagerImpl` to implement only `NotificationTemplateManager`.. or even more general terms move out from `EmailTemplateManagerImpl` by having more general notification management service.